mealie/core/security/security.py

```diff
@@ -108,16 +109,34 @@ def user_from_ldap(db: AllRepositories, username: str, password: str) -> Private
     return user
 
 
-def authenticate_user(session, email: str, password: str) -> PrivateUser | bool:
+def authenticate_user_sso(session, username: str, email: str | None, name: str | None) -> PrivateUser:
+    db = get_repositories(session)
+    user = db.users.get_one(username, "username", any_case=True)
+    if not user:
+        user = db.users.create(
+            {
+                "username": username,
+                "password": "SSO",
+                # Fill the next two values with something unique and vaguely
```

Comment:
Could this be a security risk? Could you log in through the login form using "SSO" for this user? In that would be the case, maybe a randomly generated password could add some security.